lib/ansible/plugins/inventory/online.py

```diff
@@ -0,0 +1,190 @@
+    except ValueError:
+        raise AnsibleError("Incorrect JSON payload")
+
+
+def extract_public_ipv4(host_infos):
+    try:
+        return host_infos["network"]["ip"][0]
+    except (KeyError, TypeError, IndexError):
+        return None
+
+
+def extract_private_ipv4(host_infos):
+    try:
+        return host_infos["network"]["private"][0]
+    except (KeyError, TypeError, IndexError):
+        return None
+
+
+def extract_os_name(host_infos):
+    try:
+        return host_infos["os"]["name"]
+    except (KeyError, TypeError):
+        return None
+
+
+def extract_os_version(host_infos):
+    try:
+        return host_infos["os"]["version"]
+    except (KeyError, TypeError):
+        return None
+
+
+def extract_hostname(host_infos):
+    try:
+        return host_infos["hostname"]
+    except (KeyError, TypeError):
+        return None
+
+
+extractors = {
+    "public_ipv4": extract_public_ipv4,
+    "private_ipv4": extract_private_ipv4,
+    "hostname": extract_hostname,
+}
+
+
+class InventoryModule(BaseInventoryPlugin):
+    NAME = 'online'
+
+    def _fill_host_variables(self, hostname, host_infos):
+        targeted_attributes = (
+            "offer",
+            "id",
+            "hostname",
+            "location",
+            "boot_mode",
+            "power",
+            "last_reboot",
+            "anti_ddos",
+            "hardware_watch",
+            "support"
+        )
+        for attribute in targeted_attributes:
+            self.inventory.set_variable(hostname, attribute, host_infos[attribute])
+
+        if extract_public_ipv4(host_infos=host_infos):
+            self.inventory.set_variable(hostname, "public_ipv4", extract_public_ipv4(host_infos=host_infos))
+            self.inventory.set_variable(hostname, "ansible_host", extract_public_ipv4(host_infos=host_infos))
+
+        if extract_private_ipv4(host_infos=host_infos):
+            self.inventory.set_variable(hostname, "public_ipv4", extract_private_ipv4(host_infos=host_infos))
+
+        if extract_os_name(host_infos=host_infos):
+            self.inventory.set_variable(hostname, "os_name", extract_os_name(host_infos=host_infos))
+
+        if extract_os_version(host_infos=host_infos):
+            self.inventory.set_variable(hostname, "os_version", extract_os_name(host_infos=host_infos))
+
+    def _filter_host(self, host_infos, hostname_preferences):
+
+        for pref in hostname_preferences:
+            if extractors[pref](host_infos):
+                return extractors[pref](host_infos)
+
+        return None
+
+    def do_server_inventory(self, host_infos, hostname_preferences):
+
+        hostname = self._filter_host(host_infos=host_infos,
+                                     hostname_preferences=hostname_preferences)
+
+        # No suitable hostname were found in the attributes and the host won't be in the inventory
+        if not hostname:
+            return
+
+        self.inventory.add_host(host=hostname)
```

Comment:
I see there is no grouping in this inventory. Is this intentional for the first patch? Fine to me but I'd like to raise that grouping based on `os_name` of `offer` allow greater flexibility when dealing with a fleet of machine, might be interesting to add.

Reply:
I'm working on a grouping feature that could be easily extended to geographical location and private networking subnets.

Reply:
Done for location and offer